blockchain/ethereum.go

```diff
@@ -646,6 +677,42 @@ func NewEVMClient(networkSettings EVMNetwork, env *environment.Environment) (EVM
 	for idx, networkURL := range networkSettings.URLs {
 		networkSettings.URL = networkURL
 		ec, err := newEVMClient(networkSettings)
+
+		if err != nil {
+			return nil, err
+		}
+		ec.SetID(idx)
+		ecl.Clients = append(ecl.Clients, ec)
+	}
+	ecl.DefaultClient = ecl.Clients[0]
+	wrappedClient := wrapMultiClient(networkSettings, ecl)
+	// required in Geth when you need to call "simulate" transactions from nodes
+	if ecl.NetworkSimulated() {
+		if err := ecl.Fund("0x0", big.NewFloat(1000)); err != nil {
+			return nil, err
+		}
+	}
+
+	return wrappedClient, nil
+}
+
+// ConcurrentEVMClient returns a multi-node EVM client connected to the specified network
+// It is used for concurrent interactions from different threads with the same network and from same owner
+// account. This ensures that correct nonce value is fetched when an instance of EVMClient is initiated using this method.
+func ConcurrentEVMClient(networkSettings EVMNetwork, env *environment.Environment, existing EVMClient) (EVMClient, error) {
```

Comment:
At least the beginning setup seems quite similar to `NewEVMClient`. Any way we can consolidate? Maybe we don't even need the full new method?

Reply:
Not changing this now as discussed. We are going to explore more robust method of handling concurrent tx from the same client